internal/sqldb/sqlquery/entity_variable.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlquery
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/datacommonsorg/mixer/internal/util"
+)
+
+// EntityVariable returns all existent entity, variable pairs
+func EntityVariable(sqlClient *sql.DB) (map[string]map[string]struct{}, error) {
+	defer util.TimeTrack(time.Now(), "SQL: EntityVariable")
+	query := "SELECT DISTINCT entity, variable FROM observations o"
```

Comment:
A couple questions:

1. How long does this query take on Cloud SQL for the large ONE dataset?
2. Any concerns about the size of this map? If there are 200 entities and 6000 vars, there will be 1.2M entries.

Reply:
1. 6-7 seconds
2. Give the map is keyed by <e,v> with value of an empty struct, each entry takes O(10) bytes, so for this case it's in the order 12Mb (with a factor of 5, it's 60Mb) which is a reasonable size.